client/src/java/com/zimbra/client/ZMailbox.java

```diff
@@ -933,8 +925,8 @@ public Element invoke(Element request) throws ServiceException {
     }
 
     public Element invoke(Element request, String requestedAccountId) throws ServiceException {
-        lock();
-        try {
+        try (final MailboxLock l = lockFactory.readLock()) {
```

Comment:
You could use lock() instead of directly calling the factory in all these methods.

Reply:
what do you mean?

Reply:
ZMailbox has lock() methods that use the factory.